workflows/dnaseq/dnaseq-core.wdl

```diff
@@ -0,0 +1,109 @@
+## **WARNING:** this workflow is experimental! Use at your own risk!
+
+version 1.1
+
+import "../../data_structures/read_group.wdl"
+import "../../tools/bwa.wdl"
+import "../../tools/picard.wdl"
+import "../../tools/samtools.wdl"
+import "../../tools/util.wdl"
+import "../general/bam-to-fastqs.wdl" as bam_to_fastqs_wf
+import "../general/samtools-merge.wdl" as samtools_merge_wf
+
+workflow dnaseq_core_experimental {
+    meta {
+        description: "Aligns DNA reads using bwa"
+        outputs: {
+            harmonized_bam: "Harmonized DNA-Seq BAM, aligned with bwa",
+            harmonized_bam_index: "Index for the harmonized DNA-Seq BAM file"
+        }
+        allowNestedInputs: true
+    }
+    parameter_meta {
+        read_one_fastqs_gz: "Input gzipped FASTQ format file(s) with 1st read in pair to align"
+        read_two_fastqs_gz: "Input gzipped FASTQ format file(s) with 2nd read in pair to align"
+        bwa_db: "Gzipped tar archive of the bwa reference files. Files should be at the root of the archive."
+        reads_per_file: "Controls the number of reads per FASTQ file for internal split to run BWA in parallel."
+        read_groups: "An Array of structs defining read groups to include in the harmonized BAM. Must correspond to input FASTQs. Each read group ID must be contained in the basename of a FASTQ file or pair of FASTQ files if Paired-End. This requirement means the length of `read_groups` must equal the length of `read_one_fastqs_gz` and the length of `read_two_fastqs_gz` if non-zero. Only the `ID` field is required, and it must be unique for each read group defined. See data_structures/read_group.wdl for help formatting your input JSON."
+        prefix: "Prefix for the BAM file. The extension `.bam` will be added."
+        aligner: {
+            description: "BWA aligner to use",
+            choices: ["mem", "aln"]
+        }
+        validate_input: "Ensure input BAM is well-formed before beginning harmonization?"
+        use_all_cores: "Use all cores? Recommended for cloud environments."
+    }
+    input {
+        Array[File] read_one_fastqs_gz
+        Array[File] read_two_fastqs_gz
+        File bwa_db
+        Int reads_per_file = 10000000
+        Array[ReadGroup] read_groups
+        String prefix
+        String aligner = "mem"
+        Boolean use_all_cores = false
+    }
+
+    scatter (rg in read_groups) {
+        call read_group.ReadGroup_to_string { input: read_group = rg }
+    }
+
+    Array[String] read_groups_bwa = prefix("@RG ", ReadGroup_to_string.stringified_read_group)
+
+    scatter (tuple in zip(
+        zip(read_one_fastqs_gz, read_two_fastqs_gz),
+        read_groups_bwa
+    )) {
+        call util.split_fastq as read_ones { input:
+            fastq = tuple.left.left,
+            reads_per_file = reads_per_file
+        }
+
+        call util.split_fastq as read_twos { input:
+            fastq = select_first([tuple.left.right, "undefined"]),
```

Comment:
This shouldn't be necessary if I'm understanding correctly. Can this (and later) `select_first()` be removed?